### PR TITLE
RHDM-801: improving FEEL parser 

### PR DESCRIPTION
**Pull request by etirelli**
RHDM-801: improving FEEL parser to support variable names containing the 'in' keyword everywhere except 'for' loops and in the 'every'/'some' quantifiers

### CODE REVIEW

File: kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4 [hunk 1]
```diff
@@ -176,6 +176,21 @@ nameDefinition
...
 
 nameDefinitionTokens
+    : Identifier
+        ( Identifier
+        | additionalNameSymbol
+        | IntegerLiteral
+        | FloatingPointLiteral
+        | reusableKeywords
+        | IN
+        )*
```
Comment (etirelli) [inline kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4:186]: for regular nameDefinitionTokens, I added the "IN" token as a viable alternative.

File: kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4 [hunk 2]
```diff
@@ -176,6 +176,21 @@ nameDefinition
...
+    ;
+
+iterationNameDefinition
+    : iterationNameDefinitionTokens { helper.defineVariable( $iterationNameDefinitionTokens.ctx ); }
+    ;
```
Comment (etirelli) [inline kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4:191]: created a new name definition rule exclusively for iteration contexts where it is impossible to use the IN token

File: kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4 [hunk 3]
```diff
@@ -91,8 +91,8 @@ iterationContexts
     ;
 
 iterationContext
-    : {helper.isFeatDMN12EnhancedForLoopEnabled()}? nameDefinition IN expression '..' expression
-    | nameDefinition IN expression
+    : {helper.isFeatDMN12EnhancedForLoopEnabled()}? iterationNameDefinition IN expression '..' expression
+    | iterationNameDefinition IN expression
     ;
```
Comment (etirelli) [inline kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4:96]: iteration contexts invoke a specific iterationNameDefinition rule that does not accept the IN token

File: kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4 [hunk 4]
```diff
@@ -379,7 +394,7 @@ nameRef
 
 nameRefOtherToken
     : { helper.followUp( _input.LT(1), _localctx==null ) }?
-        ~(LPAREN|RPAREN|LBRACK|RBRACK|LBRACE|RBRACE|LT|GT|EQUAL|BANG|DIV|MUL|IN|COMMA)
+        ~(LPAREN|RPAREN|LBRACK|RBRACK|LBRACE|RBRACE|LT|GT|EQUAL|BANG|DIV|MUL|COMMA)
     ;
```
Comment (etirelli) [inline kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4:398]: IN is now a valid nameRefOtherToken

File: kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4 (hunk 2 above)
Comment (tarilabs) [reply kie-dmn/kie-dmn-feel/src/main/antlr4/org/kie/dmn/feel/parser/feel11/FEEL_1_1.g4:191]: I would have added this comment as a comment in the .g4 ;)

**Review by tarilabs — APPROVED** [review]
lgtm

**Review by evacchi — APPROVED** [review]
lgtm

**Review by baldimir — APPROVED** [review]

**Pull request merged**